server/container/core/src/test/java/org/apache/james/core/MailImplTest.java

```diff
@@ -80,31 +87,173 @@ public void testConstr3() throws MessagingException {
 
         helperTestInitialState(mail);
         helperTestMessageSize(mail, 0);
-        assertEquals("initial message", mimeMessage.getMessageID(), mail.getMessage().getMessageID());
-        assertEquals("sender", sender, mail.getSender().toString());
-        assertEquals("name", name, mail.getName());
+        assertThat(mail.getMessage().getMessageID()).isEqualTo(mimeMessage.getMessageID());
+        assertThat(mail.getSender().asString()).isEqualTo(sender);
+        assertThat(mail.getName()).isEqualTo(name);
         mail.dispose();
     }
 
     @Test
-    public void testDuplicate() throws MessagingException {
+    public void duplicateShouldCreateANewJavaObject() throws MessagingException {
         MailImpl mail = new MailImpl();
         MailImpl duplicate = (MailImpl) mail.duplicate();
-        assertNotSame("is real duplicate", mail, duplicate);
+
+        assertThat(mail).as("duplicate method should return different objects").isNotSameAs(duplicate);
+
         helperTestInitialState(duplicate);
         helperTestMessageSize(duplicate, 0);
     }
 
     @Test
-    public void testDuplicateNewName() throws MessagingException {
+    public void duplicateShouldNotChangeOriginalMailName() throws MessagingException {
         String newName = "aNewName";
-
         MailImpl mail = new MailImpl();
-        assertFalse("before + after names differ", newName.equals(mail.getName()));
+        assertThat(mail.getName()).as("before + after names should differ").isNotEqualTo(newName);
```

Comment:
Don't know.

However, it was here before. There might be a reason. Even if it looks *obvious*. Thus I should not delete here in this commit (simply about test refactoring) but in another commit (like the NullPointerException)